data/vibe/data/serialization.d

```diff
@@ -571,6 +579,67 @@ private template serializeValueImpl(Serializer, alias Policy) {
 	}
 }
 
+///
+package template isSerializerSupportSinkType (SerT, ObjT)
+{
+	enum isSerializerSupportSinkType = is(typeof(SerT.serializeSinkType!(ObjT)(ObjT.init)) == void);
+}
+
+///
+package template isObjectSupportSinkType (ObjectT)
+{
+	enum isObjectSupportSinkType = is(typeof(ObjectT.toString(cast(SerializeSinkDg) null)) == void) &&
+									 is(typeof(ObjectT.fromString("")) : ObjectT);
+}
+
+///
+package alias SerializeSinkDg = void delegate(scope const(char)[]) @safe;
+
+version(unittest)
+{
+	import std.array : split;
+	import std.format : formattedWrite;
+	import vibe.data.json;
+
+	struct X (alias T)
+	{
+		private int i;
+		private string s;
+		this (int i, string s) @safe {this.i = i; this.s = s;}
```

Comment:
If it's a struct, you don't need the ctor, do you ?
If you do, missing attributes.